larray/core/array.py

```diff
@@ -1342,52 +1342,55 @@ def reindex(self, axes_to_reindex=None, new_axis=None, fill_value=np.nan, inplac
         a\\b  b0  b1
          a0   0   1
          a1   2   3
+        >>> arr2 = ndrange('a=a1,a2;c=c0;b=b2..b0')
+        >>> arr2
+         a  c\\b  b2  b1  b0
+        a1   c0   0   1   2
+        a2   c0   3   4   5
 
         Reindex one axis
 
-        >>> arr.reindex(X.b, ['b1', 'b2', 'b0'], fill_value=-1)
-        a\\b  b1  b2  b0
-         a0   1  -1   0
-         a1   3  -1   2
-        >>> arr.reindex(X.b, 'b0..b2', fill_value=-1)
+        >>> arr.reindex('b', ['b1', 'b2', 'b0'])
+        a\\b   b1   b2   b0
+         a0  1.0  nan  0.0
+         a1  3.0  nan  2.0
+        >>> arr.reindex('b', 'b0..b2', fill_value=-1)
         a\\b  b0  b1  b2
          a0   0   1  -1
          a1   2   3  -1
+        >>> arr.reindex('b', arr2.b, fill_value=-1)
+        a\\b  b2  b1  b0
+         a0  -1   1   0
+         a1  -1   3   2
+       >>> arr.reindex('b', arr2.b['b1':], fill_value=-1)
+       a\\b  b1  b0
+        a0   1   0
+        a1   3   2
 
         Reindex several axes
 
-        >>> a = Axis(['a1', 'a2', 'a0'], 'a')
-        >>> b = Axis(['b2', 'b1', 'b0'], 'b')
-        >>> arr.reindex({'a': a, 'b': b}, fill_value=-1)
+        >>> arr.reindex({'a': arr2.a, 'b': arr2.b}, fill_value=-1)
         a\\b  b2  b1  b0
          a1  -1   3   2
          a2  -1  -1  -1
-         a0  -1   1   0
-        >>> arr.reindex({X.a: a, X.b: b})
-        a\\b   b2   b1   b0
-         a1  nan  3.0  2.0
-         a2  nan  nan  nan
-         a0  nan  1.0  0.0
-
-        Reindex using axes from another array
-
-        >>> arr2 = ndrange('a=a0,a1;c=c0..c0;b=b0..b2')
-        >>> arr2
-         a  c\\b  b0  b1  b2
-        a0   c0   0   1   2
-        a1   c0   3   4   5
-        >>> arr.reindex(arr2.axes)
-         a  b\\c   c0
-        a0   b0  0.0
-        a0   b1  1.0
-        a0   b2  nan
-        a1   b0  2.0
-        a1   b1  3.0
-        a1   b2  nan
-        >>> arr2.reindex(arr.axes)
-         a  c\\b   b0   b1
-        a0   c0  0.0  1.0
-        a1   c0  3.0  4.0
+        >>> arr.reindex({'a': arr2.a['a1'], 'b': arr2.b['b1':]}, fill_value=-1)
```

Comment:
unsure showing reindexing to a single label is worth it. It is likely to confuse people more than it helps. I would scrap this example from the docstring. It is an interesting unit test for several reasons though.